### PR TITLE
Flavorize xfce4-session

### PR DESCRIPTION
**Pull request by OlivierDuchateau**
Signed-off-by: Olivier Duchateau <duchateau.olivier@gmail.com>

Flavorize x11-wm/xfce4-session in order to install easily x11/xscreensaver or x11/xfce4-screensaver.

- default (without support of x11/xscreensaver)
- xscreensaver

Adjust the metaport (x11-wm/xfce4), by default it is x11-wm/xfce4-session@xscreensaver (like previous releases).

### CODE REVIEW

**Comment by madpilot78** [comment]
While I don't dislike the idea I can't accept this change because it would be refused by portmgr.

Flavors are for use when the port is built in different ways resulting, for example, in different binaries. (think binary linked with libX11 and friends for X11 support and not linked with any of those without X11 UI).

Subpackages, a planned functionality for the ports tree, still unavailable, would be perfect in this case. That functionality is for when the same compiled port can be installed in different ways and the only difference is in plist.

Due to this the rule is that flavors can't be leveraged to get subpackages like functionality. Since any port adding flavors support requires explicit portmgr approval, this patch would be refused such approval.

An acceptable way to do this would be to make what you are doing with flavors in an option, and create a slave port, to include in the meta port. Also, instead of duplicating the plist, PLIST_SUB(and thee related option helpers) should be used.

Since I'm looking at other details now, could you modify this pull request in thee way I indicated?

File: x11-wm/xfce4/Makefile [hunk 1]
```diff
@@ -2,23 +2,25 @@
...
-		xfsettingsd:sysutils/xfce4-settings
+		xfsettingsd:sysutils/xfce4-settings \
 
 USES=	metaport
 
-OPTIONS_DEFINE=		APPFINDER GREYBIRD GTK2ENGINE ICONS LIGHTDM MOUSEPAD NOTIFY ORAGE PRINT SQUEEZE \
+OPTIONS_DEFINE=	APPFINDER GREYBIRD GTK2ENGINE ICONS LIGHTDM MOUSEPAD NOTIFY ORAGE PRINT SQUEEZE \
 			TERMINAL XPM
-OPTIONS_DEFAULT=	APPFINDER GREYBIRD ICONS MOUSEPAD NOTIFY TERMINAL
+OPTIONS_SINGLE=		SAVER
```
Comment (madpilot78) [inline x11-wm/xfce4/Makefile]: Also this is incorrect.

the option for XSCREENSAVER is in x11/xfce4-goodies, so this option group should be there.

As is this patch would create a duplicate. But the logic is that anything that is not core xfce should go in goodies.

**Comment by madpilot78** [comment]
Maybe there is another way to attack this issue.

Choosing which screensaver to use should be a runtime decision, not one to be made at installation time. Both a flavor (or subpackage) and a slave port look inadequate to me.

This is more a user setting than a system package issue.

**Comment by OlivierDuchateau** [comment]
> Maybe there is another way to attack this issue.
> 
> Choosing which screensaver to use should be a runtime decision, not one to be made at installation time. Both a flavor (or subpackage) and a slave port look inadequate to me.
> 
> This is more a user setting than a system package issue.

I don't know the best way (flavor or subpackage, and so on), the goal is to avoid to install Xscreensaver autostart desktop file, when we use x11/xfce4-screensaver (I find it is confusing).

**Comment by madpilot78** [comment]
> I don't know the best way (flavor or subpackage, and so on), the goal is to avoid to install Xscreensaver autostart desktop file, when we use x11/xfce4-screensaver (I find it is confusing).

But it's not causing any errors or failures, correct?

If that's the case and it's only a minor annoyance it can be fixed a later time.

I'd really like to get XFCE 4.14 into the tree without causing too much confusion. having a different package for the screen saver is going to be confusing for users anyway.

OPTIONS are only used when building from source We have no statistics about how many users are building from source and how many using packages, but the project is pushing for users to move to packages.

Al;so, there is nothing stopping a user from installling xscreensaver manually (not through the meta port) and then he would be missing a necessary .desktop file.

**Comment by OlivierDuchateau** [comment]
You can close this pull request, I think it's better to keep ports as simple as possible. You are right, anyone can install manually any screensaver.

**Pull request closed without merging**